eng/mgmt/automation/generate.py

```diff
@@ -70,40 +72,73 @@ def generate(
     update_version(sdk_root, service)
 
     if compile:
-        if os.system(
-                'mvn clean verify package -f {0}/pom.xml -pl {1}:{2} -am'.
-                format(sdk_root, GROUP_ID, module)) != 0:
+        if os.system('mvn clean verify package -f {0}/pom.xml -pl {1}:{2} -am'.
+                     format(sdk_root, GROUP_ID, module)) != 0:
             logging.error('[GENERATE] Maven build fail')
             return False
 
     return True
 
 
+def add_module_to_modules(modules: str, module: str) -> str:
+    post_module = re.search(r'([^\S\n\r]*)</modules>', modules)
+    indent = post_module.group(1)
+    indent += '  '
+
+    all_module = set(re.findall(r'<module>(.*?)</module>', modules))
+    all_module.add(module)
+    all_module = [
+        indent + POM_MODULE_FORMAT.format(module)
+        for module in sorted(all_module)
+    ]
+
+    return '<modules>\n' + ''.join(all_module) + post_module.group()
+
+
+def add_module_to_default_profile(pom: str, module: str) -> (bool, str):
+    for profile in re.finditer(r'<profile>[\s\S]*?</profile>', pom):
+        profile_value = profile.group()
+        if re.search(r'<id>default</id>', profile_value):
+            if len(re.findall('<modules>', profile_value)) > 1:
+                logging.error(
+                    '[POM][Profile][Skip] find more than one <modules> in <profile> default'
+                )
+                return (False, '')
+            modules = re.search(r'<modules>[\s\S]*</modules>', profile_value)
+            if not modules:
+                logging.error(
+                    '[POM][Profile][Skip] Cannot find <modules> in <profile> default'
+                )
+                return (False, '')
+            modules_update = add_module_to_modules(modules.group(), module)
+            pre_modules = pom[:profile.start() + modules.start()]
+            post_modules = pom[profile.start() + modules.end():]
+            return (True, pre_modules + modules_update + post_modules)
+    logging.error(
+        '[POM][Profile][Skip] cannot find <profile> with <id> default')
+    return (False, '')
+
+
 def add_module_to_pom(pom: str, module: str) -> (bool, str):
     if pom.find('<module>{0}</module>'.format(module)) >= 0:
         logging.info('[POM][Skip] pom already has module {0}'.format(module))
-        return (False, '')
+        return (True, pom)
 
     if len(re.findall('<modules>', pom)) > 1:
+        if pom.find('<profiles>') >= 0:
+            return add_module_to_default_profile(pom, module)
```

Comment:
Do you also need to add it to coverage profile?

Reply:
Will us run coverage later?

Reply:
Since you added it in template above (for new service pom.xml).

Or use similar one in our resourcemanager folder? Not sure whether this is for bypassing the coverage.
```
        <dependency>
          <groupId>com.azure</groupId>
          <artifactId>azure-sdk-template</artifactId>
          <version>1.2.1-beta.16</version> <!-- {x-version-update;com.azure:azure-sdk-template;current} -->
        </dependency>
```

Reply:
I do not really know what it would do. The parameter is just copied from https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/mediaservices/pom.xml

If we don't need the coverage, I think we don't care about it.

Reply:
I mean, for current code, when this is new service, you put the artifact into dependency of coverage profile (and default); when this is not a new service, you do not put the artifact into dependency of coverage profile (only to default). Would better to be consistent?

Reply:
This add <module> to a pom that already has other module (data-plane), not for the new pom. (Not sure whether you mistake it)

Reply:
Could you add sample pom to clarify, 1. for new service (no data-plane); 2. for service already having data-plane?

Reply:
coverage removed